Docs: recommendation for packaging uberjars (#14292)
apache/iceberg: site/docs/multi-engine-support.md

When using Iceberg with these engines, the runtime jar is the only addition to the classpath needed in addition to vendor dependencies.
For example, to use Iceberg with Spark 3.5 and AWS integrations, `iceberg-spark-runtime-3.5_2.12` and AWS SDK dependencies are needed for the Spark installation.

> > ℹ️ It's important to make sure that only the runtime jars (plus storage specific bundles if needed, eg. `iceberg-aws-bundle` or `iceberg-gcp-bundle`) are included in the runtime classpath.

Reviewer: how does this page render? Can you provide a screenshot when running the site locally?

> All other modules should be excluded as they may introduce dependency version conflicts with engine runtimes.

> > For example, when packaging uberjar for a Spark application, we should include `iceberg-spark-runtime` and exclude modules such as `iceberg-core` or `iceberg-parquet`.

Reviewer: ```suggestion
> For example, when packaging an uberjar for a Spark application, only include `iceberg-spark-runtime` and exclude modules such as `iceberg-core` or `iceberg-parquet`.
```
